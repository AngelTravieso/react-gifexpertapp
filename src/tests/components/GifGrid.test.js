import React from 'react';
import { GifGrid } from '../../components/GifGrid';
import { shallow } from 'enzyme';
import { useFetchGifs } from '../../hooks/useFetchGifs';

jest.mock('../../hooks/useFetchGifs');

describe('Pruebas en <GifGrid />', () => {

    const category = 'Bleach';
    
    test('debe mostrarse correctamente ', () => {

        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        });
        
        const wrapper = shallow( <GifGrid category={ category } /> );
        expect( wrapper ).toMatchSnapshot();

    });


    test('debe de mostrar items cuando se cargan imagenes useFetch', () => {

        const gifs = [{
            id: 'ABC',
            url: 'https:://localhost/prueba.jpg',
            title: 'prueba'
        }];

        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        });

       
        const wrapper = shallow( <GifGrid category={ category } /> );
        expect( wrapper ).toMatchSnapshot();

    });
    
    


});
