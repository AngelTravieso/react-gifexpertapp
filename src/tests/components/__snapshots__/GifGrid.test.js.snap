// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGrid /> debe de mostrar items cuando se cargan imagenes useFetch 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    Bleach
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="prueba"
      url="https:://localhost/prueba.jpg"
    />
  </div>
</Fragment>
`;

exports[`Pruebas en <GifGrid /> debe mostrarse correctamente  1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    Bleach
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
